infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -111,11 +128,22 @@ public SenderBufferManager(DataSender dataSender) {
         msgTimer = DefaultClusterConfig.getLogSenderResendTimer();
         timeoutTimer = DefaultClusterConfig.getLogSenderTimeoutTimer();
         errorOnMsgTimeout = DefaultClusterConfig.isLogSenderTimeout();
+        maxLogEntryWait = DefaultClusterConfig.getLogSenderWaitPeriod();
 
         readConfig();
         pendingMessages = new SenderPendingMessageQueue(maxBufferSize);
         pendingCompletableFutureForAcks = new HashMap<>();
         this.dataSender = dataSender;
+
+        // Back of napkin math to derive sleep time (per message):
+        // Assume payload size of 15MB
+        // Queue holds ~50 messages before OOM risk
+        // maxSleepTime (default) = 5min to process 50 messages = 300000 / 50 = 6000 + buffer = 7000ms
```

Comment:
is 5mins an arbitrary number for the max wait across 50 messages?

Reply:
Not completely arbitrary, it was the time needed to apply the transactions the 20mb entry from your initial triage, but also not concrete which is why I've went ahead and made that parameter tunable.